src/dfx-core/src/fs/mod.rs

```diff
@@ -18,6 +18,7 @@ pub fn canonicalize(path: &Path) -> Result<PathBuf, FsError> {
 }
 
 pub fn copy(from: &Path, to: &Path) -> Result<u64, FsError> {
+    create_dir_all(&parent(to)?)?;
```

Comment:
The methods here are meant as simple wrappers for methods in std::fs.  Methods that do more would go in the `composite` module and have a different name.

Also, if there are only certain places where the target directory might not exist, I'd rather call that out with an explicit call to `dfx_core::fs::composite::ensure_parent_dir_exists`.